go/vt/wrangler/reparent.go

```diff
@@ -795,14 +795,14 @@ func (maxPosSearch *maxReplPosSearch) processTablet(tablet *topodatapb.Tablet) {
 	maxPosSearch.maxPosLock.Unlock()
 }
 
-// chooseNewMaster finds a tablet that is going to become master after reparent. The criteria
+// ChooseNewMaster finds a tablet that is going to become master after reparent. The criteria
 // for the new master-elect are (preferably) to be in the same cell as the current master, and
 // to be different from avoidMasterTabletAlias. The tablet with the largest replication
 // position is chosen to minimize the time of catching up with the master. Note that the search
 // for largest replication position will race with transactions being executed on the master at
 // the same time, so when all tablets are roughly at the same position then the choice of the
 // new master-elect will be somewhat unpredictable.
-func (wr *Wrangler) chooseNewMaster(
+func (wr *Wrangler) ChooseNewMaster(
```

Comment:
Why is this function exported? Do we expect to use it outside wrangler?

Reply:
Yes, we did, although after some offline discussion with @enisoc, we decided it made more sense to write a public method on Wrangler that returns all candidates that are fully caught up, which `chooseNewMaster` would use. That's what I've done with my latest commit that's now pushed up.